dot/node.go

```diff
@@ -298,6 +300,13 @@ func NewNode(cfg *Config, ks *keystore.GlobalKeystore, stopFunc func()) (*Node,
 		publishMetrics(cfg)
 	}
 
+	gd, err := stateSrvc.Storage.GetGenesisData()
+	if !cfg.Global.NoTelemetry && err == nil {
```

Comment:
add an err check case before this?

Reply:
Do you mean `if err == nil && !cfg.Global.NoTelemetry {`  or put the err check in a separate block?  I have them together because both must be true before I can add that endpoint connection.

Reply:
yeah i mean add an err check case before:
```
gd, err := stateSrvc.Storage.GetGenesisData()
if err != nil {
    return nil, err
}

if cfg.Global.NoTelemetry {
    return node, nil 
}

// logic inside current block here, since we know  !cfg.Global.NoTelemetry && err == nil is true at this point
```

Reply:
Ok, yes this is more readable.